### PR TITLE
HUE-9509 [ui] Create AlertBox component

### PR DESCRIPTION
**Pull request by sreenaths**
https://issues.cloudera.org/browse/HUE-9509

### CODE REVIEW

File: desktop/core/src/desktop/js/components/AlertBox.vue [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+  export enum AlertType {
+    Success,
+    Error,
+    Warning,
+    Info,
+    Unknown
+  }
+
+  class TypeDefinition {
```
[the remark this answers is not in the document]
Comment (JohanAhlen) [reply desktop/core/src/desktop/js/components/AlertBox.vue:57]: Could also be just an interface, and then [AlertType.Success, { title: I18n('Success') ... }] below

File: desktop/core/src/desktop/js/components/AlertBox.vue [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    @Prop({ default: false })
+    showClose!: boolean;
+
+    showDetails = false;
+
+    get definition(): TypeDefinition {
+      return TYPE_DEFS.get(this.type) || UNKNOWN_DEF;
+    }
+
+    setShowDetails(show: boolean): void {
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/js/components/AlertBox.vue:97]: Nit: no need for the function as can do showDetails = x above, or change to toggleDetails(): void

File: desktop/core/src/desktop/js/components/AlertBox.vue [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    <em :class="['fa alert-icon alert-color', definition.faClass]" />
+    <div class="alert-box-title">{{ definition.title }}</div>
+
+    <span class="alert-box-message">
+      {{ message }}
+    </span>
+
+    <div v-if="showDetails">
+      <a v-if="details" class="more-less-button" @click="setShowDetails(false)">
+        Less details
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/js/components/AlertBox.vue:32]: {{ I18n('Less details') }} and add methods: { I18n } to `@component`

File: desktop/core/src/desktop/js/components/AlertBox.vue [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      </a>
+      <pre class="alert-box-details">
+        {{ details }}
+      </pre>
+    </div>
+    <a v-else-if="details" class="more-less-button" @click="setShowDetails(true)">
+      More details
```
Comment (JohanAhlen) [inline desktop/core/src/desktop/js/components/AlertBox.vue:39]: I18n...

**Review by JohanAhlen — APPROVED** [review]
Nice!

File: desktop/core/src/desktop/js/components/AlertBox.vue (hunk 2 above)
Comment (sreenaths) [reply desktop/core/src/desktop/js/components/AlertBox.vue]: showDetails = x looks sweet 😍

**Pull request merged**